allow independent-agent-release-staging for dra staging workflow

gogochan opened the pull request:
This PR allows [independent-agent-release-staging](https://buildkite.com/elastic/independent-agent-release-staging) to trigger Stack installers Staging workflow.

Independent Agent Release needs to include Windows installer. Our intent is to only include Agent binary.

gogochan committed:
allow independent-agent-release-staging for dra staging workflow

dliappis reviewed (commented):
We should also adjust the conditional in https://github.com/elastic/elastic-stack-installers/blob/b67ab1c118e092139f1344443ddbe6da8048e02a/.buildkite/scripts/build.ps1#L58 to also exclude `independent-agent-release-staging`

gogochan committed:
add independent-agent-pattern in build.ps1

gogochan committed:
temporary change for testing

gogochan committed:
close bracket

gogochan committed:
line continuation

gogochan committed:
remove line continuation

amitkanfer committed:
Adding support for Agent version with 'build' information

amitkanfer commented:
build failed, but seems unrelated to the recent changes?

gogochan commented:
Yeah, this seems to be unrelated. I've posted a question in https://github.com/elastic/elastic-stack-installers/issues/282#issuecomment-2215032278

dliappis committed:
Also trim + suffixes from VERSION

With elastic agent early releases we have a version scheme like
8.14.0+build202407021002 . This commit adjusts the existing filter
for VERSION such that it can also trim +.* suffixes.
